R/compare-opts.R

```diff
@@ -52,3 +58,12 @@ old_opts <- function(..., tol, check.attributes, checkNames) {
 
   out
 }
+
+object_opts <- function(x) {
+  attr(x, "waldo_opts")
+}
+
+merge_opts <- function(...) {
+  allopts <- zap_nulls(list(...))
+  purrr::reduce(allopts, utils::modifyList)
```

Comment:
Yeah, fine to use `Reduce()`. I don't think it's worth taking a dependency on purrr for just one function.

Reply:
Done.